src/armlsd.py

```diff
@@ -60,6 +60,7 @@ def ARMLSD(argflag, spect, fitsdict, reuseMaster=False):
         slf = sciexp[sc]
         scidx = slf._idx_sci[0]
         msgs.info("Reducing file {0:s}, target {1:s}".format(fitsdict['filename'][scidx], slf._target_name))
+        msgs.sciexp = slf  # For QA writing on exit, if nothing else.  Could write Masters too
```

Comment:
I think I have it.

Check it out.  Not super elegant, but effective.
Could be made more elegant if need be.

> On Apr 19, 2016, at 9:14 AM, Ryan Cooke <notifications@github.com> wrote:
> 
> In src/armlsd.py <https://github.com/PYPIT/PYPIT/pull/43#discussion_r60261811>:
> 
> > @@ -60,6 +60,7 @@ def ARMLSD(argflag, spect, fitsdict, reuseMaster=False):
> >          slf = sciexp[sc]
> >          scidx = slf._idx_sci[0]
> >          msgs.info("Reducing file {0:s}, target {1:s}".format(fitsdict['filename'][scidx], slf._target_name))
> > +        msgs.sciexp = slf  # For QA writing on exit, if nothing else.  Could write Masters too
> I'm not aware of a way to get sciexp in the traceback... For now, I'm happy to merge your approach, but we will need to revisit this issue. If you decide to merge, be sure to raise an issue so that we come back to this. Especially needed for when an unexpected bug is encountered (and not just an error message).
> … <x-msg://178/#>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> 
> !DSPAM:826,571659851602718217784! <https://github.com/PYPIT/PYPIT/pull/43#discussion_r60260623> <https://github.com/PYPIT/PYPIT/pull/43> <https://github.com/PYPIT/PYPIT/pull/43#discussion_r60258460> <http://matplotlib.org/examples/pylab_examples/multipage_pdf.html> <https://github.com/PYPIT/PYPIT/pull/43/files/ce9c3eeb928cf7ad173485d0e4fbab1a901d7798#r60258460> <http://www.ucolick.org/~xavier/> <http://imps.ucolick.org/> <http://www.qpqsurvey.org/> <http://k1dm3.ucolick.org/> <http://www.ucolick.org/~xavier/IDL/index.html> <https://github.com/pyigm/pyigm> <https://github.com/linetools/linetools> <https://github.com/PYPIT/PYPIT/pull/43/files/ce9c3eeb928cf7ad173485d0e4fbab1a901d7798#r60260623>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/PYPIT/PYPIT/pull/43/files/ce9c3eeb928cf7ad173485d0e4fbab1a901d7798#r60261811>
> !DSPAM:826,571659851602718217784!

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/pyigm/pyigm
https://github.com/linetools/linetools

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)